### PR TITLE
#6350: Add mem config for TTNN operations

### LINKED ISSUE #6350: Some ttnn operations are missing memory_config parameter
Most ttnn operations are having `memory_config: ttnn.MemoryConfig = ttnn.DRAM_MEMORY_CONFIG` parameter. For example:

https://tenstorrent-metal.github.io/tt-metal/latest/ttnn/ttnn/signbit.html

But some operations are missing it. Like:

```
permute
reshape
rmsnorm
repeat_interleave
global_avg_pool2d
split
```

### PR DESCRIPTION
**Pull request by Aswinmcw**
Tracking Issue: #6350 

### Checklist
- [x] Post commit CI passes - https://github.com/tenstorrent/tt-metal/actions/runs/10317795787
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/pybind11/operations/core.hpp
Comment (KalaivaniMCW) [inline ttnn/cpp/pybind11/operations/core.hpp:1]: all occurrences of ttnn::reshape now passing a memory_config right ? 
should it be added in generic_reductions.hpp as well ?
based on the files listed here https://github.com/search?q=repo%3Atenstorrent%2Ftt-metal%20ttnn%3A%3Areshape&type=code ?

Comment (Aswinmcw) [reply ttnn/cpp/pybind11/operations/core.hpp:1]: Yeah i have added in that file

**Review by KalaivaniMCW — APPROVED** [review]
Approved

File: ttnn/cpp/pybind11/operations/core.hpp [hunk 1]
```diff
@@ -19,51 +19,57 @@ namespace core {
 void py_module(py::module& module) {
     module.def(
         "reshape",
-        [](const ttnn::Tensor& tensor, const ttnn::Shape& shape) -> ttnn::Tensor {
-            return ttnn::reshape(tensor, shape);
+        [](const ttnn::Tensor& tensor, const ttnn::Shape& shape, const std::optional<MemoryConfig>& memory_config) -> ttnn::Tensor {
+            return ttnn::reshape(tensor, shape, memory_config);
```
Comment (sjameelTT) [inline ttnn/cpp/pybind11/operations/core.hpp:23]: why do we need memory_config conversions in reshape? What's the use-case?

Comment (Aswinmcw) [reply ttnn/cpp/pybind11/operations/core.hpp:23]: It has been raised as an issue here
https://github.com/tenstorrent/tt-metal/issues/6350
So I have added it

**Review by eyonland — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/core/core.cpp [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 namespace ttnn::operations::core {
 
-ttnn::Tensor reshape(const ttnn::Tensor& tensor, const ttnn::Shape& shape) {
+ttnn::Tensor reshape(const ttnn::Tensor& tensor, const ttnn::Shape& shape, const std::optional<MemoryConfig>& memory_config) {
```
Comment (tt-aho) [inline ttnn/cpp/ttnn/operations/core/core.cpp:16]: This arg seems unused for this fn so why do we need it added here?

Comment (eyonland) [reply ttnn/cpp/ttnn/operations/core/core.cpp:16]: No reason for it that I can see.

File: ttnn/cpp/pybind11/operations/core.hpp (hunk 1 above)
Comment (ayerofieiev-tt) [reply ttnn/cpp/pybind11/operations/core.hpp:23]: @nemanjagrujic @sjameelTT can you please resolve this?

Comment (nemanjagrujic) [reply ttnn/cpp/pybind11/operations/core.hpp:23]: Hello, the ticket reports and we noticed that some ttnn operations miss memory_config parameter while almost all of the operations have it (And lists the operations where the parameter is missing). 

If that behaviour is expected (and not by mistake), that's fine and can be responded in ticket comments.

**Comment by Aswinmcw** [comment]
Closing as memory config parameter is not needed for these ops

**Pull request closed without merging**